src/ldp.rs

```diff
@@ -1133,6 +1180,133 @@ impl ProofSuite for Eip712Signature2021 {
+        let sig_hex = crate::keccak_hash::bytes_to_lowerhex(sig_bytes);
+        proof.proof_value = Some(sig_hex);
+        Ok(proof)
+    }
+
+    async fn prepare(
+        &self,
+        document: &(dyn LinkedDataDocument + Sync),
+        options: &LinkedDataProofOptions,
+        _public_key: &JWK,
+    ) -> Result<ProofPreparation, Error> {
+        let proof = Proof {
+            context: serde_json::json!([EPSIG_CONTEXT.clone()]),
+            proof_purpose: options.proof_purpose.clone(),
+            verification_method: options.verification_method.clone(),
+            created: Some(options.created.unwrap_or_else(now_ms)),
+            domain: options.domain.clone(),
+            challenge: options.challenge.clone(),
+            ..Proof::new("EthereumPersonalSignature2021")
+        };
+        let signing_string = string_from_document_and_options(document, &proof).await?;
+        Ok(ProofPreparation {
+            proof,
+            jws_header: None,
+            signing_input: SigningInput::EthereumPersonalMessage {
+                ethereum_personal_message: signing_string,
+            },
+        })
+    }
+
+    async fn complete(
+        &self,
+        preparation: ProofPreparation,
+        signature: &str,
+    ) -> Result<Proof, Error> {
+        let mut proof = preparation.proof;
+        proof.proof_value = Some(signature.to_string());
+        Ok(proof)
+    }
+
+    async fn verify(
+        &self,
+        proof: &Proof,
+        document: &(dyn LinkedDataDocument + Sync),
+        resolver: &dyn DIDResolver,
+    ) -> Result<(), Error> {
+        let sig_hex = proof
+            .proof_value
+            .as_ref()
+            .ok_or(Error::MissingProofSignature)?;
+        let verification_method = proof
+            .verification_method
+            .as_ref()
+            .ok_or(Error::MissingVerificationMethod)?;
+        let vm = resolve_vm(&verification_method, resolver).await?;
+        match &vm.type_[..] {
+            "EcdsaSecp256k1VerificationKey2019" => (),
+            "EcdsaSecp256k1RecoveryMethod2020" => (),
+            _ => Err(Error::VerificationMethodMismatch)?,
+        };
+        if !sig_hex.starts_with("0x") {
+            return Err(Error::HexString);
+        }
+        let dec_sig = hex::decode(&sig_hex[2..])?;
+        let rec_id = k256::ecdsa::recoverable::Id::try_from(dec_sig[64] - 27)?;
+        let sig = k256::ecdsa::Signature::try_from(&dec_sig[..64])?;
+        let sig = k256::ecdsa::recoverable::Signature::new(&sig, rec_id)?;
+        let signing_string = string_from_document_and_options(document, &proof).await?;
+        let hash = crate::keccak_hash::hash_personal_message(&signing_string);
+        let digest = k256::elliptic_curve::FieldBytes::<k256::Secp256k1>::from_slice(&hash);
+        let recovered_key = sig.recover_verify_key_from_digest_bytes(&digest)?;
+        // TODO this step needs keccak-hash, may need better features management
```

Comment:
which step? You're already doing the hash manually above

Reply:
Fixed in 6bb6a39